### DISCUSSION TITLE
add mouse_filter attribute for canvas layer that applies on all child control node

### DISCUSSION THREAD

**Discussion by melonlireal** [godotengine/godot-proposals#10289] [GUI]
version: 4.2
Since most people use a canvas layer to contain controll nodes. This feature could be very helpful. Forexample, if I have two sets of UI under two canvas layer in the same scene (game UI and setting UI). If I want to switch between UI and makesure the other set of UI don't react to input. I must set the mouse_filter attribute for all controll nodes under the disabled canvas layer one by one. By giving canvas layer the mouse filter attribute, it will only take one line of code.

(please lmk if there are eaiser method for implementing this UI switch system with the current version though)

**Comment by asheden** [comment]
they're gonna reply any day now

**Comment by kitbdev** [comment]
As of 4.5, [mouse_behavior_recursive](https://docs.godotengine.org/en/stable/classes/class_control.html#class-control-property-mouse-behavior-recursive) can be used for this on any Control.
